Automattic/wp-calypso
# Site Profiler: Extract the hosting provider details to a separate hook

Related to https://github.com/Automattic/wp-calypso/issues/91778

## Proposed Changes

Introduce a new hook that extracts the logic related to the hosting provider details.

## Why are these changes being made?


 This allows for the logic to be re-used by future implementations like in https://github.com/Automattic/wp-calypso/issues/91778.

## Testing Instructions



* Go to http://calypso.localhost:3000/setup/hosted-site-migration/
* Enter a URL for a WordPress not hosted by A8C, for example - https://minlovecat.sg/
* On the next screen, select a destination or choose "Skip and create a new site".
* On the next screen, as before, you should see a paragraph with the hosting provider's name.

![image](https://github.com/Automattic/wp-calypso/assets/1612178/89c88773-02c8-4a85-8856-55613cbcc9be)


## Pre-merge Checklist
